src/cli/audit_command.zig

```diff
@@ -0,0 +1,570 @@
+            }
+            entry.value_ptr.dependents.deinit();
+        }
+        self.vulnerable_packages.deinit();
+        self.all_vulnerabilities.deinit();
+    }
+};
+
+pub const AuditCommand = struct {
+    pub fn exec(ctx: Command.Context, pm: *PackageManager, args: [][:0]u8) !void {
+        _ = args;
+        Output.prettyError(comptime Output.prettyFmt("<r><b>bun pm audit <r><d>v" ++ Global.package_json_version_with_sha ++ "<r>\n", true), .{});
+        Output.flush();
+
+        const load_lockfile = pm.lockfile.loadFromCwd(pm, ctx.allocator, ctx.log, true);
+        @import("./package_manager_command.zig").PackageManagerCommand.handleLoadLockfileErrors(load_lockfile, pm);
+
+        var dependency_tree = try buildDependencyTree(ctx.allocator, pm);
+        defer dependency_tree.deinit();
+
+        const packages_json = try collectPackagesForAudit(ctx.allocator, pm);
+        defer ctx.allocator.free(packages_json);
+
+        const response_text = try sendAuditRequest(ctx.allocator, pm, packages_json);
+        defer ctx.allocator.free(response_text);
+
+        if (response_text.len > 0) {
+            try printEnhancedAuditReport(ctx.allocator, response_text, pm, &dependency_tree);
+        } else {
+            Output.prettyln(comptime Output.prettyFmt("<green>No vulnerabilities found.<r>", true), .{});
+        }
+    }
+};
+
+fn buildDependencyTree(allocator: std.mem.Allocator, pm: *PackageManager) !std.StringHashMap(std.ArrayList([]const u8)) {
+    var dependency_tree = std.StringHashMap(std.ArrayList([]const u8)).init(allocator);
+
+    const packages = pm.lockfile.packages.slice();
+    const pkg_names = packages.items(.name);
+    const pkg_dependencies = packages.items(.dependencies);
+    const pkg_resolutions = packages.items(.resolutions);
+    const buf = pm.lockfile.buffers.string_bytes.items;
+    const dependencies = pm.lockfile.buffers.dependencies.items;
+    const resolutions = pm.lockfile.buffers.resolutions.items;
+
+    for (pkg_names, pkg_dependencies, pkg_resolutions, 0..) |pkg_name, deps, res_list, pkg_idx| {
+        const package_name = pkg_name.slice(buf);
+
+        if (packages.items(.resolution)[pkg_idx].tag != .npm) continue;
+
+        const dep_slice = deps.get(dependencies);
+        const res_slice = res_list.get(resolutions);
+
+        for (dep_slice, res_slice) |_, resolved_pkg_id| {
+            if (resolved_pkg_id >= pkg_names.len) continue;
+
+            const resolved_name = pkg_names[resolved_pkg_id].slice(buf);
+
+            const result = try dependency_tree.getOrPut(try allocator.dupe(u8, resolved_name));
+            if (!result.found_existing) {
+                result.value_ptr.* = std.ArrayList([]const u8).init(allocator);
+            }
+            try result.value_ptr.append(try allocator.dupe(u8, package_name));
+        }
+    }
+
+    return dependency_tree;
+}
+
+fn collectPackagesForAudit(allocator: std.mem.Allocator, pm: *PackageManager) ![]u8 {
```

Comment:
```suggestion
fn collectPackagesForAudit(allocator: std.mem.Allocator, pm: *PackageManager) bun.OOM![]u8 {
```
restricting error sets can make more readable code